dali/util/image.h

```diff
@@ -91,18 +91,20 @@ typedef int (*outFunc)(const vector<uint8_t> &tmp, int h, int w, int c,
  * @brief Writes an image after applying a scale and bias to get
  * pixel values in the range 0-255
  */
-template <typename T>
+template <typename Backend, typename T>
 void WriteImageScaleBias(const T *img, int h, int w,
     int c, float bias, float scale, const string &file_name, outFunc pFunc) {
   DALI_ENFORCE(img != nullptr);
   DALI_ENFORCE(h >= 0);
   DALI_ENFORCE(w >= 0);
   DALI_ENFORCE(c >= 0);
-  CUDA_CALL(cudaDeviceSynchronize());
 
   vector<uint8_t> cpu_vector(h * w * c, 0);
-  MemCopy(cpu_vector.data(), img, cpu_vector.size(), 0);
-  CUDA_CALL(cudaStreamSynchronize(0));
+  if (std::is_same<Backend, GPUBackend>::value) {
+    MemCopy(cpu_vector.data(), img, cpu_vector.size(), 0);
```

Comment:
you removed the stream synchronize here. Was it intentional?

Also, why using std::copy for CPU backend? Was there any problem?

Reply:
Because this function can be called for CPU only backend. So no cuda related calls inside.
Also MemCopy calls cudaMemcpyAsync under the hood. If you don't want to relay on lincuda.so you should not use it.